docs/timezone.md

```diff
@@ -45,6 +47,51 @@ tz1.getTransitions(now).next().done;  // => true
 tz2.getTransitions(now).next().done;  // => false
 ```
 
+## Temporal.TimeZone.from(thing: string | object) : Temporal.TimeZone
+
+**Parameters:**
+- `thing` (string or object): A `Temporal.TimeZone` object or a string from which to create a `Temporal.TimeZone`.
+
+**Returns:** a new `Temporal.TimeZone` object.
+
+This static method creates a new time zone from another value.
+If the value is a string, it can be:
+- a string that is accepted by `new Temporal.TimeZone()`;
+- a string in the ISO 8601 format including a time zone part, optionally extended with an IANA time zone name in square brackets appended to it;
+- or, a string with _only_ the time zone part of the ISO 8601 format, again extended with optional IANA name in brackets.
+Or, if the value is an object, it can be another `Temporal.TimeZone` object, which is returned directly.
```

Comment:
What about a non-Temporal.TimeZone object with the right properties, in analogy to `Temporal.Date.from({year: 2020, month: 1, day: 23})` and `Temporal.Duration.from({hours: 24})`?

Reply:
The only property is `name`, so you'd have `Temporal.TimeZone.from({name: 'America/Vancouver'})` which seems a bit redundant with `Temporal.TimeZone.from('America/Vancouver')`. However, it seems reasonable to add this capability for symmetry, if people feel strongly about it.

Reply:
If the only property is `name`, then we have a separate problem of time zone objects not being constructible. Isn't a Temporal.TimeZone object intended to represent a [tzdata](https://data.iana.org/time-zones/tz-how-to.html) entity?

Reply:
Not sure if I understand what you mean with time zone objects not being constructible? If you mean that an object can have a UTC offset and no name, then maybe that means that the `name` property is misnamed; since `Temporal.TimeZone.from('-08:00').name === '-08:00'` so presumably if we allowed construction from an options bag we'd allow `{ name: '-08:00' }`.

Reply:
No, I mean that time zone objects consist fundamentally of a collection of rules describing when and how local time transitions between (UTC) offsets. Although a time zone can be _looked up_ from a common database by name, it's not a name—it's a data structure, and constructing a time zone from scratch is an important use case (cf. [`getTimeZoneObjectFromRules`](https://github.com/tc39/proposal-temporal/issues/240#user-content-timezone-instance), among others).

Reply:
Constructing "synthetic" time zone objects is an API that we describe neither in the spec nor in the polyfill right now. I'd consider that out of scope for fixing the current problems with `TimeZone.from()`, but I opened another issue for it: #332

Reply:
Given the undeveloped state of those objects, I'm fine with keeping them out of scope for this PR.